src/box/journal.h

```diff
@@ -188,19 +188,12 @@ void
 journal_queue_wakeup(void);
 
 /**
- * Check whether any of the queue size limits is reached.
- * If the queue is full, we must wait for some of the entries to be written
- * before proceeding with a new asynchronous write request.
+ * Yield until there's some space in the journal queue.
+ *
+ * If no_wait is true then fail if request has to yield.
  */
-static inline bool
-journal_queue_is_full(void)
-{
-	return journal_queue.size >= journal_queue.max_size;
-}
-
-/** Yield until there's some space in the journal queue. */
 int
-journal_queue_wait(struct journal_entry *entry);
+journal_queue_wait(struct journal_entry *entry, bool no_wait);
```

Comment:
The API of sort `wait(bool no_wait)` looks not really good. Please, lets find a way to check for the potential waiting explicitly in the calling code.

Like add a method like `journal_queue_would_block()` which checks for the size + request queue being non-empty.

Alternatively I could suggest to add a new enum for `journal.h` like `enum journal_submit_mode { JOURNAL_SUBMIT_MODE_WAIT, JOURNAL_SUBMIT_MODE_NO_WAIT; }` instead of a flag and you would then translate txn wait mode into the journal wait mode. This magic true/false in the code is really bothering.

Reply:
These solutions looks more heavy to me, especially the second. So I'll try the first one.